apps/01PigLatin.js

```diff
@@ -11,35 +11,90 @@ var rl = readline.createInterface({
 function pigLatin(word) {
 
   // Your code here
+  //first test
+  // var word /= 'car'; unecessary to define this
 
-}
+  word = word.toLowerCase();
+
+  var firstLetter = word[0];
+  console.log('first letter:', firstLetter);
+
+  var vowelIndex = -1; //set to assume no vowls are found
+
+  // we can 'remove' letters by replace them with and empty string
+  console.log(word.replace('a', ''));
+
+  //next test ~~
+
+  console.log('crazy'.indexOf('a'));
+
+  //if it doesn't find it, it will return -1 look through strings for an integer
+  console.log('crazy'.indexOf('e'));
+
+  if ((word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('a');
+  }
+
+  if ((word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('e');
+  }
 
+  if ((word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('i');
+  }
+  if ((word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('o');
+  }
+  if ((word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('u');
+  }
+  if ((word.indexOf('y') > -1 && word.indexOf('y') < vowelIndex) || vowelIndex === -1) {
+    vowelIndex = word.indexOf('y');
+  }
+  console.log('hello'.slice(0, 2));
+
+  // notice that it returns up until the second index given (before 'l' at index 2)
+
+  var firstPart = word.slice(0, vowelIndex);
+  var restWord = word.slice(vowelIndex, word.length);
+  console.log('first part', firstPart)
+  console.log('rest word', restWord)
+
+  // next test begins with vowels
+
+  if (vowelIndex === 0) {
+    return word + 'yay';
+  } else {
+    return restWord + firstPart + 'ay';
+  }
+
+
```

Comment:
Nice work! Good to see that you're making use of `console.log()` to check on how data is changing throughout your function.